src/promptflow/promptflow/executor/flow_executor.py

```diff
@@ -1210,12 +1210,20 @@ def enable_streaming_for_llm_flow(self, stream_required: Callable[[], bool]):
         :return: None
         """
         for node in self._flow.nodes:
+            streaming_option_parameter = self._parse_streaming_option_parameter(node)
             if (
-                self._flow.is_llm_node(node)
+                streaming_option_parameter is not None
                 and self._flow.is_referenced_by_flow_output(node)
                 and not self._flow.is_referenced_by_other_node(node)
             ):
-                self._tools_manager.wrap_tool(node.name, wrapper=_inject_stream_options(stream_required))
+                wrapper = _inject_stream_options(stream_required, streaming_option_parameter)
+                self._tools_manager.wrap_tool(node.name, wrapper=wrapper)
+
+    def _parse_streaming_option_parameter(self, node: Node) -> Optional[str]:
+        if self._flow.is_llm_node(node):
+            return "stream"
```

Comment:
Shall we make this as a fallback logic only when `tool_function._streaming_option_parameter` is None?